packages/type-utils/src/typeOrValueSpecifiers/typeDeclaredInPackage.ts

```diff
@@ -0,0 +1,55 @@
+import * as ts from 'typescript';
+
+function findParentModuleDeclaration(
+  node: ts.Node,
+): ts.ModuleDeclaration | undefined {
+  switch (node.kind) {
+    case ts.SyntaxKind.ModuleDeclaration:
+      return node as ts.ModuleDeclaration;
+    case ts.SyntaxKind.SourceFile:
+      return undefined;
+    default:
+      return findParentModuleDeclaration(node.parent);
+  }
+}
+
+function typeDeclaredInDeclareModule(
+  packageName: string,
+  declarations: ts.Node[],
+): boolean {
+  return declarations.some(
+    declaration =>
+      findParentModuleDeclaration(declaration)?.name.text === packageName,
+  );
+}
+
+function typeDeclaredInDeclarationFile(
```

Comment:
Might want to make this function's name a bit more clear that it is for package declaration files.

e.g. `typeDeclaredInPackageDeclarationFile`